app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.update({self.name: self})
+
+
+def create_person_list(people1: list) -> list:
+    output_list = []
+    for pers in people1:
+        pers_inst = Person(pers["name"], pers["age"])
+        if "wife" in pers and pers["wife"] is not None:
+            pers_inst.wife = pers["wife"]
+        elif "husband" in pers and pers["husband"] is not None:
+            pers_inst.husband = pers["husband"]
```

Comment:
Firstly, you need just create all instances of Person without assign a husband and a wife. You can add the created instances to output_list. And the last, you can use a list comprehension instead of the loop.

Reply:
Still not fixed